specs/protocol/abi.md

```diff
@@ -553,16 +553,18 @@ _Important note:_ For the JSON representation of receipts, we represent 64-bit u
 - `type`: `Panic`.
 - `id`: Hexadecimal string representation of the 256-bit (32-byte) contract ID of the current context if in an internal context. `null` otherwise.
 - `reason`: Decimal string representation of an 8-bit unsigned integer; panic reason.
+- `data`: Hexadecimal string representation of the 256-bit (32-byte) contract ID or asset ID related to the error. `null` otherwise.
```

Comment:
Instead of `null`, could this field be made optional?

Reply:
My concern about `null` is it is nouveau to the specs. The serialization of this new variant should probably be a boolean byte with a flag on whether or not the field is there, and the following bytes optionally with the contents.

I would also avoid creating fields that might contain different meanings. In this case, we should probably have two optional fields - one for the contract id, the other for the asset id. Having the information mixed might lead to complications downstream.

Reply:
`null` is used elsewhere though, see `id` above.

Reply:
True; interestingly enough, we still serialize the (zeroed) fields in the canonical serialization:

https://github.com/FuelLabs/fuel-tx/blob/0cb216da4fa2a823caa402f1973594cda4406f9b/src/receipt/receipt_std.rs#L75-L83

If we just extend the same strategy then we should be fine. I don't think this discrepancy between abi and memory serialization will ever be an issue

Reply:
Updated this to `contractId`, we can add `assetId` later if actually needed.